packages/agent/src/utils.ts

```diff
@@ -84,4 +84,89 @@ export async function getPaginationCursor(message: RecordsWriteMessage, dateSort
 
 export function webReadableToIsomorphicNodeReadable(webReadable: ReadableStream<any>) {
   return new ReadableWebToNodeStream(webReadable);
-}
\ No newline at end of file
+}
+
+export function appendPathToUrl({ path, url }: { url: string, path: string }): string {
+  const urlObject = new URL(url);
+  const lastChar = urlObject.pathname.slice(-1);
+
+  if (lastChar === '/') {
+    urlObject.pathname += path;
+  } else {
+    urlObject.pathname += `/${path}`;
+  }
+
+  return urlObject.toString();
+}
+
+/**
+ * Polling function with interval, TTL accepting a custom fetch function
+ * @template T - the return you expect from the fetcher
+ * @param fetchFunction an http fetch function
+ * @param [interval=3000] how frequently to poll
+ * @param [ttl=300_000] how long until polling stops
+ * @returns T - the result of fetch
+ */
+export function pollWithTTL(
```

Comment:
Convention:

`pollWithTtl`, yeah, it's kinda ugly in this case but convention wins, consider `pollUntilTimeout` instead. In fact, `dwn-sdk-js` has exported a very similar method called `pollUntilSuccessOrTimeout` if you are interested.

Reply:
done